packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -56,25 +68,34 @@ const variantsResolver = (props, styles, theme, name) => {
 
 const shouldForwardProp = (prop) => prop !== 'styleProps' && prop !== 'theme';
 
-const muiStyled = (tag, options, muiOptions) => {
+const experimentalStyled = (tag, options, muiOptions = {}) => {
   const name = muiOptions.muiName;
   const defaultStyledResolver = styled(tag, { shouldForwardProp, label: name, ...options });
   const muiStyledResolver = (...styles) => {
-    if (muiOptions.overridesResolver) {
-      styles.push((props) => {
-        const theme = props.theme || defaultTheme;
+    const stylesWithDefaultTheme = styles.map((stylesArg) => {
+      return typeof stylesArg === 'function'
+        ? ({ theme: themeInput, ...rest }) =>
+            stylesArg({ theme: isEmpty(themeInput) ? defaultTheme : themeInput, ...rest })
```

Comment:
discussion: That seems like a surprising behavior for defaultizing. Usually you fall back to a default when using `undefined`, maybe `null`. But I'd be surprised if some code would use a default value when passed `{}`.

Reply:
Both emotion & styled-components default their theme object to empty object instead of null/undefined, that is why this check is necessary - https://codesandbox.io/s/frosty-violet-pxkre?file=/src/App.js 

I spend quite some time figuring out why our theme was not considered until I found out this 🤷